### PR TITLE
Issue/3020 fix exports with group 1 repeatable

### PR DESCRIPTION
**Pull request by stellanl**
#### Before the PR (what is the issue or what needed to be done)
Forms w all repeatable groups and responses with >100 repetitions resulted in broken reports.
#### The solution
Handle it, by writing one row at a time.
## Checklist
* [ ] Test plan
* [ ] Copyright header
* [ ] Code formatting
* [ ] Documentation

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 1]
```diff
@@ -529,80 +532,96 @@ private void analyseData(final List<InstanceData> allData) {
...
+        createCell(r, col++, dto.getKeyId().toString());
+        createCell(r, col++, ExportImportUtils.formatDateTime(dto.getCollectionDate()));
+        createCell(r, col++, sanitize(dto.getSubmitterName()));
+        String duration = getDurationText(dto.getSurveyalTime());
+        createCell(r, col++, duration);
+        Double v = dto.getFormVersion();
+        if (v != null) {
+            createCell(r, col++, v);
+        }
+        //TODO else a cell with 0.0??
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: Is this still needed?

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 2]
```diff
@@ -654,18 +689,21 @@ private synchronized int writeInstanceData(
         }
 
         // Calculate the digest
+        /* only for split sheets
+        //TODO this may go outside the window!
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: Also still needed?

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 3]
```diff
@@ -529,80 +532,96 @@ private void analyseData(final List<InstanceData> allData) {
...
+            Row iterationRow = getRow(startRow + i, sheet);
+            writeMetadataRow(iterationRow, instanceData, i + 1, groupSheet);
+            for (QuestionDto qd : whichQuestions) {
+                final Long questionId = qd.getKeyId();
+                final QuestionDto questionDto = questionsById.get(questionId);
+                SortedMap<Long, String> iterationsMap = instanceData.responseMap.get(questionId);
+                if (iterationsMap == null) {
+                    continue; //No answers on this sheet
+                }
+                String val = iterationsMap.get(new Long(i));
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: It feels weird to have a `new Long()` here.  Maybe its worth it converting that and related code to an `int`?

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java (hunk 2 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: No

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java (hunk 1 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: Might be. We have not had any feedback on the form version column.

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java (hunk 3 above)
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: Then we cannot use a Map, it depends on the mapping being Object->Object

File: GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java [hunk 4]
```diff
@@ -529,80 +532,96 @@ private void analyseData(final List<InstanceData> allData) {
...
-                writeAnswer(iterationRow, columnIndexMap.get(questionId.toString()),
-                        questionDto, val);
-                if (!digestRows.contains(iterationRow)) { //A Set would be neater
-                    digestRows.add(iterationRow);
+        for (int i = 0; i < maxIter; i++) {
+            Row iterationRow = getRow(startRow + i, sheet);
+            writeMetadataRow(iterationRow, instanceData, i + 1, groupSheet);
+            for (QuestionDto qd : whichQuestions) {
+                final Long questionId = qd.getKeyId();
+                final QuestionDto questionDto = questionsById.get(questionId);
```
[the remark this answers is not in the document]
Comment (stellanl) [reply GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java]: Ah, cut-and-paste consequence

**Comment by stellanl** [comment]
The starting row problem was that when there were NO questions on the base sheet, there was nothing entered for that instance. Now the code always puts in a row for each surveyInstance, even if it will be only metadata.

**Pull request merged**